### PR TITLE
dynamicly quantized linear benchmarking

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30148 dynamicly quantized linear benchmarking**

Differential Revision: [D18613006](https://our.internmc.facebook.com/intern/diff/D18613006)

### CODE REVIEW

File: benchmarks/operator_benchmark/pt/qlinear_test.py [hunk 1]
```diff
@@ -45,17 +47,31 @@ def init(self, N, IN, OUT):
         qW = torch.quantize_per_tensor(W, scale=scale, zero_point=0, dtype=torch.qint8)
 
         self.input = qX
-        self.qlinear = nnq.Linear(IN, OUT)
+
+        # Assume that the `self.qlinear` is set in the child
         self.qlinear.weight = qW
         self.qlinear.scale = scale
         self.qlinear.zero_point = zero_point
```
Comment (jianyuh) [inline benchmarks/operator_benchmark/pt/qlinear_test.py]: I have a uncommitted PR: https://github.com/pytorch/pytorch/pull/24059.

For qconv2d op, we have scale and zero points as torch.tensor (https://github.com/pytorch/pytorch/blob/master/benchmarks/operator_benchmark/pt/qconv_test.py#L75-L76). However, for qlinear op, we have these quantization parameters as the pure number.

There are some back-and-forth changes here. Please advise what is the current status. I see at least there is mismatch between qlinear and qconv.

**Review by jianyuh — APPROVED** [review]
The PR part looks good to me. Please address my comment. Thanks!

File: benchmarks/operator_benchmark/pt/qlinear_test.py (hunk 1 above)
Comment (jianyuh) [reply benchmarks/operator_benchmark/pt/qlinear_test.py]: The benchmark code will not have the unit test. So it is likely any code you push here for this file will not have the unit test coverage (Correct me if I am wrong). If so, please make sure you can locally run these benchmarks.

Comment (z-a-f) [reply benchmarks/operator_benchmark/pt/qlinear_test.py]: I think for the linear it is still accepting the non-tensor variables

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit 9e48bd8:
* **1/1** broken upstream at merge base 7a8261e (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=7a8261e962f26f84f6211bca4270713a3102146c&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=9e48bd84eb589304b8ee0f940a75f6fcb038b7b6).

### 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_macos_10_13_py3_test](https://dr.pytorch.org/build-details.html?build_id=72727789)



---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

**Pull request closed without merging**